python/pyspark/pandas/groupby.py

```diff
@@ -895,6 +895,89 @@ def sem(col: Column) -> Column:
             bool_to_numeric=True,
         )
 
+    # TODO: 1, 'n' accepts list and slice; 2, implement 'dropna' parameter
+    def nth(self, n: int) -> FrameLike:
```

Comment:
https://github.com/apache/spark/blob/5a03f70aa1c6f709bee6c7d91e4e74bf38498b5a/python/pyspark/sql/functions.py#L3622

Since 3.1, there are a `def nth_value` in spark, but considering negetive index and we are going to support list and slice in the future, I think use `row_number` is right in here, but just FYI if you have other idea.

Reply:
I guess we can not apply `nth_value` for this purpose, it return the `n-th` row within one partition for each input row, can not use it to filter out unnecessary rows.